src/hooks/useTargetClusterNamespaces.ts

```diff
@@ -0,0 +1,52 @@
+import * as k8s from '@kubernetes/client-node';
+
+import {Dispatch, SetStateAction, useEffect, useRef, useState} from 'react';
+
+import {useAppSelector} from '@redux/hooks';
+import {getTargetClusterNamespaces} from '@redux/services/resource';
+
+import NamespaceHandler from '@src/kindhandlers/Namespace.handler';
+
+export const ALL_NAMESPACES = '<all>';
+export const NO_NAMESPACE = '<none>';
+
+export function useTargetClusterNamespaces(options: {
+  extra?: ('all' | 'none' | 'default')[];
+}): [string[], Dispatch<SetStateAction<string[]>>] {
+  const context = useAppSelector(state => state.config.kubeConfig.currentContext);
+  const kubeconfigPath = useAppSelector(state => state.config.kubeconfigPath);
+
+  const optionsExtra = useRef<string[]>([]);
+  optionsExtra.current = options.extra || [];
+
+  const [namespaces, setNamespaces] = useState<string[]>([]);
+
+  useEffect(() => {
+    const setClusterNamespaces = async () => {
+      const kc = new k8s.KubeConfig();
+      kc.loadFromFile(kubeconfigPath);
+      kc.setCurrentContext(context || '');
+
+      const ns = await NamespaceHandler.listResourcesInCluster(kc);
```

Comment:
let's move this and the 3 lines above into the `getTargetClusterNamespaces` function